docs/01_introduction/hardware.md

```diff
@@ -0,0 +1,112 @@
+# Choosing the Optimal Computer
+
+Though Python is runnable on most of modern operating systems (OS) including Windows, MacOS and Linux, it is beneficial to keep scripting under *nix environment. Here we provide a guide for beginners to choose your computing hardwares.
+
+This guide is intentionally written for programming beginners to code locally. For advance research units equip with python servers, we will cover a series of remote coding techniques to unleash more complex bioimage analysis.
+
+## General Guide 
+
+When choosing a computer for bioimage analysis, it's essential to consider hardware performance, memory size, OS, portability, and application scenarios. Here’s a summary comparing different computing modalities:
+
+| Feature                   | Laptops                                  | Desktops                                 | Workstations                             | Servers                                   |
+|---------------------------|------------------------------------------|------------------------------------------|------------------------------------------|------------------------------------------|
+| **Hardware Performance**  | Mid-range to high-end CPUs and GPUs      | High-end CPUs and GPUs                   | Top-tier CPUs and multiple GPUs          | Multiple high-end CPUs and GPUs          |
+| **Memory Size**           | Up to 64GB (most have 16GB-32GB)         | Up to 128GB or more                      | 128GB to 512GB or more                   | Terabytes of RAM                         |
+| **GPU**                   | Integrated or dedicated GPUs | High-end dedicated GPUs (e.g., NVIDIA RTX) | Professional GPUs (e.g., NVIDIA Quadro/RTX A-series) | Multiple professional GPUs (e.g., NVIDIA Tesla/Quadro) |
+| **NPU**                   | SoC Dependent                | Limited NPU support                      | Available in some high-end models        | Available, especially in AI-optimized servers |
+| **OS**                    | Windows, macOS, Linux                    | Windows, macOS, Linux                    | Windows, macOS, Linux                           | Linux, Windows Server                    |
+| **Portability**           | Highly portable                          | Not portable                             | Not portable                             | Not portable                             |
+| **Application Scenarios** | Mobile work, basic to moderate tasks     | Stationary use, moderate to intensive tasks | Intensive tasks, advanced analysis       | Large-scale projects, remote access, collaborative research |
+| **ARM vs x86**            | Mostly x86 (some ARM options like Apple Silicon and Snapdragon XLite) | Mostly x86 except for Apple                               | Mostly x86 except for Apple                               | Mostly x86 (ARM servers available, e.g., AWS Graviton) |
```

Comment:
I'm curious: Have you tried installing Pyton on a non-Mac ARM computer, e.g. featuring the Snapdragon CPU. I'm curious how well the Python ecosystem is compatible with these machines. Also which Operating System runs on non-Mac ARM computers?

Reply:
So far I have tested on several ARM platforms like Raspberry Pi and nvidia SBC. Both are running under Ubuntu.

Software slowly catching up with pre-compiled libraries and Linux side. If one using VSCode and miniforge the environment is quite mature. 

miniforge has no pre-compiled version for ARM Windows. 100% not recommended.

Certain image libraries that require OpenGL, and some SoC from Broadcom (RPI) run natively OpenGLES2 that emulates OpenGL using mesa drivers from Debian. High-performance rendering like 3D data plots is very bottlenecked.

Reply:
ok, then how likely is it that someone who aims at analysing images uses a Raspberry PI or an NVidia SBC?